### PR TITLE
Uninitialized scalar variable `&temp_data`

### LINKED ISSUE #68636: [Coverity CID: 344506] Uninitialized scalar variable in subsys/mgmt/mcumgr/grp/img_mgmt_client/src/img_mgmt_client.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/b8556d0d79770825952c13ddadc4782707163e2f/subsys/mgmt/mcumgr/grp/img_mgmt_client/src/img_mgmt_client.c

Category: Uninitialized variables
Function: `upload_message_header_size`
Component: Other
CID: [344506](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=344506)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/b8556d0d79770825952c13ddadc4782707163e2f/subsys/mgmt/mcumgr/grp/img_mgmt_client/src/img_mgmt_client.c#L271

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=344506

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_UNINIT). ([CWE-457](http://cwe.mitre.org/data/definitions/457.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### PR DESCRIPTION
**Pull request by ghost**
`zcbor_map_start_encode`function is using Uninitialized variable `&temp_data`

Fixes #68636

### CODE REVIEW

File: subsys/mgmt/mcumgr/grp/img_mgmt_client/src/img_mgmt_client.c [hunk 1]
```diff
@@ -266,7 +266,8 @@ static size_t upload_message_header_size(struct img_gr_upload *upload_state)
 	} else {
 		map_count = 10;
 	}
-
+	
+	temp_data = NULL;
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/grp/img_mgmt_client/src/img_mgmt_client.c:270]: Move to line 258 as part of the variable setup

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: subsys/mgmt/mcumgr/grp/img_mgmt_client/src/img_mgmt_client.c (hunk 1 above)
Comment (nu1l-ptr) [reply subsys/mgmt/mcumgr/grp/img_mgmt_client/src/img_mgmt_client.c]: Thanks, I moved it to line 258.

File: subsys/mgmt/mcumgr/grp/img_mgmt_client/src/img_mgmt_client.c [hunk 2]
```diff
@@ -266,7 +266,7 @@ static size_t upload_message_header_size(struct img_gr_upload *upload_state)
 	} else {
 		map_count = 10;
 	}
-
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/grp/img_mgmt_client/src/img_mgmt_client.c:269]: Can you undo this change?

**Review by nordicjm — COMMENTED** [review]
The commits need to be squashed into one, if you do a git rebase then mark the other 2 as `fixup`. Also the commit title and message should be formatted as per https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-guidelines

**Comment by nu1l-ptr** [comment]
I think i need to use `memset(temp_data, 0, sizeof(temp_data))` instead of `temp_data = NULL` because it is giving `incompatible pointer to integer conversion assigning to 'uint8_t'` error.

**Comment by de-nordic** [comment]
> I think i need to use `memset(temp_data, 0, sizeof(temp_data))` instead of `temp_data = NULL` because it is giving `incompatible pointer to integer conversion assigning to 'uint8_t'` error.

The `temp_data` ain't pointer.

**Review by de-nordic — COMMENTED** [review]
Click in the comment https://github.com/zephyrproject-rtos/zephyr/pull/68591#issuecomment-1928939854:
image

File: subsys/mgmt/mcumgr/grp/img_mgmt_client/src/img_mgmt_client.c [hunk 3]
```diff
@@ -266,7 +266,8 @@ static size_t upload_message_header_size(struct img_gr_upload *upload_state)
 	} else {
 		map_count = 10;
 	}
-	temp_data = NULL;
+
+	memset(temp_data, 0, sizeof(temp_data));
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/grp/img_mgmt_client/src/img_mgmt_client.c:270]: I don't really know what you are doing? Remove this and change line 258 to:
```
	uint8_t temp_data = 0;
```

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: subsys/mgmt/mcumgr/grp/img_mgmt_client/src/img_mgmt_client.c (hunk 3 above)
Comment (nu1l-ptr) [reply subsys/mgmt/mcumgr/grp/img_mgmt_client/src/img_mgmt_client.c]: sorry, I just not too use to git.

**Review by de-nordic — CHANGES_REQUESTED** [review]
Sorry, need to introduce yourself when doing changes:
https://docs.zephyrproject.org/latest/contribute/guidelines.html#signed-off-by
image

**Review by nordicjm — CHANGES_REQUESTED** [review]
Title needs to be more defined e.g. `mgmt: mcumgr: grp: img_mgmt_client: <blah>` and missing commit message

**Comment by carlescufi** [comment]
@nu1l-ptr we need this fix for the upcoming 3.6.0 release. Please fix the issues or we will need to open a parallel Pull Request fixing this. Thanks in advance!

Please take a look at [our commit message guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-guidelines) to find out how to format your commit messages, and at [our contribution workflow](https://docs.zephyrproject.org/latest/contribute/guidelines.html#contribution-workflow) to understand how to update your Pull Request.

**Comment by henrikbrixandersen** [comment]
> @nu1l-ptr we need this fix for the upcoming 3.6.0 release. Please fix the issues or we will need to open a parallel Pull Request fixing this. Thanks in advance!
> 
> Please take a look at [our commit message guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-guidelines) to find out how to format your commit messages, and at [our contribution workflow](https://docs.zephyrproject.org/latest/contribute/guidelines.html#contribution-workflow) to understand how to update your Pull Request.

Superseded by https://github.com/zephyrproject-rtos/zephyr/pull/68974
